ios/WebSocketClient.swift

```diff
@@ -68,6 +68,24 @@ class WebSocketClient: RCTEventEmitter, WebSocketDelegate {
         WebSocketManager.default.disconnectAll()
     }
 
+    override func invalidate() {
+        WebSocketManager.default.invalidateContext()
+    }
+
+    @objc(ensureClientFor:withOptions:withResolver:withRejecter:)
+    func ensureClientFor(urlString: String, options: Dictionary<String, Any> = [:], resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
+        guard let url = URL(string: urlString) else {
+            rejectMalformed(url: urlString, withRejecter: reject)
+            return
+        }
+
+        if WebSocketManager.default.getWebSocket(for: url) != nil {
+            WebSocketManager.default.invalidateClient(for: url)
```

Comment:
I preferred the conservative approach. We don't know what happened to get into this state, so we are not sure how old this client is. Also, a new set of options may have been sent. So I guessed it is better to fall on the safe case of invalidating the client than use a potentially malformed client.

Reply:
The we should be doing the same for android, no?

Reply:
It is, but not so obvious. In `andriod/.../WebsocketClientModule.kt` lines 56-59 we are doing the invalidation. This piece of code is the same as in the invalidate function. I did not use here the invalidate function since that one requires the promise, and will resolve or reject it, and that is not what I wanted. I could have extracted that piece of code to a different function, so this is more obvious.